debugging/vov-bug-finder/kp_view_of_views_bug_finder.cpp

```diff
@@ -0,0 +1,175 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <kp_core.hpp>
+
+#include <cassert>
+#include <cstdint>
+#include <cstdlib>
+#include <iostream>
+#include <map>
+#include <mutex>
+#include <optional>
+#include <string>
+
+namespace {
+
+bool verbose        = false;
+bool abort_on_error = true;
+
+class {
+  uint64_t count_;
+  std::map<uint64_t, std::string> map_;
+
+ public:
+  std::mutex mutex;
+  uint64_t push(std::string s) {
+    auto it = map_.emplace_hint(map_.end(), count_, std::move(s));
+    assert(++it == map_.end());
+    return count_++;
+  }
+  void pop(uint64_t x) {
+    auto it = map_.find(x);
+    assert(it != map_.end());
+    map_.erase(it);
+  }
+  std::string const &top() {
+    assert(!map_.empty());
+    return map_.begin()->second;
+  }
+  bool is_empty() noexcept { return map_.empty(); }
+} current;
+
+bool ignore_fence(std::string_view s) {
+  return (s == "Kokkos::Impl::ViewValueFunctor: View init/destroy fence") ||
+         (s == "Kokkos::ThreadsInternal::fence: Unnamed Instance Fence");
+}
+
+bool ignore_alloc(std::string_view s) {
+  // TODO replace poor man's starts_with and ends_with when C++20 is available
+  return (s.find("Kokkos::") == 0 &&
+          s.rfind("::scratch_mem") == s.length() - 13);
```

Comment:
This will not find the Threads backend: 

```
deallocating "Kokkos::thread_scratch" within parallel region "Z4mainE3$_0"
Abort trap: 6
```

```suggestion
  return (s.find("Kokkos::") == 0);
```